public/js/selectors/charts.js

```diff
@@ -0,0 +1,142 @@
+import moment from "moment";
+import {
+    compareIssueDates,
+    getComparisonTimeSeriesFromResponses
+} from "../helpers/chartsHelpers";
+import { createSelector } from "reselect";
+
+const getCharts = ({ charts }) => charts;
+
+const getMoments = ({ startDate, endDate }) => ({
+    startDate: moment(startDate),
+    endDate: moment(endDate)
+});
+
+/* Composer vs Incopy */
+
+const getRawComposerVsIncopy = createSelector(
+    getCharts,
+    ({ composerVsInCopy }) => composerVsInCopy
+);
+
+const getComposerVsIncopyData = createSelector(
+    getRawComposerVsIncopy,
+    composerVsInCopy => {
+        const {
+            chartData: { composerResponse, inCopyResponse },
+            pending
+        } = composerVsInCopy;
+
+        if (pending) {
+            return {
+                absolute: [],
+                percent: []
+            };
+        }
+
+        const { startDate, endDate } = getMoments(composerVsInCopy);
+
+        return getComparisonTimeSeriesFromResponses(
+            composerResponse,
+            inCopyResponse,
+            startDate,
+            endDate,
+            "Created in composer",
+            "Created in InCopy"
+        );
+    }
+);
+
+export const getComposerVsIncopy = createSelector(
+    getRawComposerVsIncopy,
+    getComposerVsIncopyData,
+    ({ error, isStacked }, data) => ({
+        data,
+        error,
+        isStacked
+    })
+);
+
+/* In Workflow vs not in Workflow */
+
+const getRawInWorkflowVsNotInWorkflow = createSelector(
+    getCharts,
+    ({ inWorkflowVsNotInWorkflow }) => inWorkflowVsNotInWorkflow
+);
+
+const getInWorkflowVsNotInWorkflowData = createSelector(
+    getRawInWorkflowVsNotInWorkflow,
+    inWorkflowVsNotInWorkflow => {
+        const {
+            chartData: { inWorkflowResponse, notInWorkflowResponse },
+            pending,
+        } = inWorkflowVsNotInWorkflow;
+    
+        if (pending) {
+            return {
+                absolute: [],
+                percent: []
+            };
+        }
+
+        const { startDate, endDate } = getMoments(inWorkflowVsNotInWorkflow);
+    
+        return getComparisonTimeSeriesFromResponses(
+            inWorkflowResponse,
+            notInWorkflowResponse,
+            startDate,
+            endDate,
+            "In Workflow",
+            "Never in Workflow"
+        );
+    }
+);
+
+export const getInWorkflowVsNotInWorkflow = createSelector(
+    getRawInWorkflowVsNotInWorkflow,
+    getInWorkflowVsNotInWorkflowData,
+    ({ error, isStacked }, data) => ({
+        data,
+        error,
+        isStacked
+    })
+);
+
```

Comment:
I'd either put a comment in here to indicate it's the start of the fork time functions or remove the other comments.